src/Widgets/DisplayWidget.vala

```diff
@@ -153,4 +174,19 @@ public class Network.Widgets.DisplayWidget : Gtk.Grid {
             break;
         }
     }
+
+    private void update_vpn_connection () {
+        active_vpn_connection = null;
+
+        nm_client.get_active_connections ().foreach ((connection) => {
+            if (active_vpn_connection == null && connection.vpn) {
+                active_vpn_connection = (NM.VpnConnection) connection;
+
+                vpn_revealer.reveal_child = active_vpn_connection.get_vpn_state () == NM.VpnConnectionState.ACTIVATED;
+                active_vpn_connection.vpn_state_changed.connect (() => {
+                    vpn_revealer.reveal_child = active_vpn_connection.get_vpn_state () == NM.VpnConnectionState.ACTIVATED;
+                });
```

Comment:
We need to disconnect it also on state change, but I wonder if this case won't be already covered by `nm_client.notify["active-connections"]`

Reply:
There are other states an "active" VPN could be in like error states or waiting for password states etc, so this is a pretty dumb indicator in that it only shows if the "active" connection is also "ACTIVATED"